lib/Backend/arm64/Thunks.asm

```diff
@@ -26,40 +26,36 @@
     ;Js::Var NativeCodeGenerator::CheckCodeGenThunk(Js::RecyclableObject* function, Js::CallInfo callInfo, ...)
     NESTED_ENTRY ?CheckCodeGenThunk@NativeCodeGenerator@@SAPEAXPEAVRecyclableObject@Js@@UCallInfo@3@ZZ
 
-    PROLOG_SAVE_REG_PAIR fp, lr, #-(16*8+32)!          ; establish stack frame
-    PROLOG_SAVE_REG_PAIR x19, x20, #16
-    stp     x0, x1, [sp, #32+0*8]
-    stp     x2, x3, [sp, #32+2*8]
-    stp     x4, x5, [sp, #32+4*8]
-    stp     x6, x7, [sp, #32+6*8]
-    stp     x8, x9, [sp, #32+8*8]
-    stp     x10, x11, [sp, #32+10*8]
-    stp     x12, x13, [sp, #32+12*8]
-    stp     x14, x15, [sp, #32+14*8]
+    PROLOG_SAVE_REG_PAIR fp, lr, #-(2*8+16*8)!          ; establish stack frame
+    stp     d0, d1, [sp, #16+0*8]
+    stp     d2, d3, [sp, #16+2*8]
+    stp     d4, d5, [sp, #16+4*8]
+    stp     d6, d7, [sp, #16+6*8]
+    stp     x0, x1, [sp, #16+8*8]
+    stp     x2, x3, [sp, #16+10*8]
+    stp     x4, x5, [sp, #16+12*8]
+    stp     x6, x7, [sp, #16+14*8]
 
     bl      |?CheckCodeGen@NativeCodeGenerator@@SAP6APEAXPEAVRecyclableObject@Js@@UCallInfo@3@ZZPEAVScriptFunction@3@@Z|  ; call  NativeCodeGenerator::CheckCodeGen
+    mov     x15, x0               ; move entry point to x15
 
 #if defined(_CONTROL_FLOW_GUARD)
     mov     x15, x0               ; __guard_check_icall_fptr requires the call target in x15
```

Comment:
>mov     x15, x0 [](start = 4, length = 15)

Is this mov redundant?

Reply:
Ah, I see you removed it in the second commit.

---
In reply to: [152667204](https://github.com/Microsoft/ChakraCore/pull/4294#discussion_r152667204) [](ancestors = 152667204)